metriq_gym/job_manager.py

```diff
@@ -1,9 +1,8 @@
-from dataclasses import asdict
+from dataclasses import asdict, dataclass
```

Comment:
[comment]: It would probably be better to use pydantic dataclasses if we can, but I recall them causing some odd linting/mypy issues for certain things (which may be hinting at a broader issue, but we can leave that out-of-scope for now).

Reply:
Indeed, mypy is complaining about `asdict` when used with pydantic dataclasses ([this issue](https://github.com/python/mypy/issues/17242)), and starting from this PR I wanted mypy actively checking this file too.